test/parallel/xyk-pallet.L1asset.register.update.test.ts

```diff
@@ -0,0 +1,249 @@
+/*
+ *
+ * @group asset
+ * @group parallel
+ */
+import { jest } from "@jest/globals";
+import { getApi, initApi } from "../../utils/api";
+import { Keyring } from "@polkadot/api";
+import { User } from "../../utils/User";
+import { api, getSudoUser, setupApi, setupUsers } from "../../utils/setup";
+import {
+  getEventErrorFromSudo,
+  getEventResultFromMangataTx,
+} from "../../utils/txHandler";
+import { ExtrinsicResult } from "../../utils/eventListeners";
+import { Sudo } from "../../utils/sudo";
+import { Assets } from "../../utils/Assets";
+import { randomBytes } from "crypto";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(1500000);
+process.env.NODE_ENV = "test";
+
+//let testUser1: User;
+let sudo: User;
+
+let keyring: Keyring;
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  keyring = new Keyring({ type: "ethereum" });
+
+  sudo = getSudoUser();
+  keyring.addPair(sudo.keyRingPair);
+
+  await setupApi();
+});
+
+test("Asset can be created by a sudo user", async () => {
+  const assetId = await api.query.tokens.nextCurrencyId();
+  const tokenEthereumAddress = "0x" + randomBytes(20).toString("hex");
+  await sudo.registerL1Asset(assetId, tokenEthereumAddress).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  const idToL1Asset = JSON.parse(
+    JSON.stringify(await api.query.assetRegistry.idToL1Asset(assetId)),
+  );
+  expect(idToL1Asset.ethereum).toEqual(tokenEthereumAddress);
+});
+
+test("Asset can't be created by a regular user", async () => {
+  const [testUser] = setupUsers();
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser, Assets.DEFAULT_AMOUNT),
+  );
+
+  const assetId = await api.query.tokens.nextCurrencyId();
```

Comment:
same :)

Reply:
also assetId is optional, so you don't really need to guess it before registering. 
![image](https://github.com/mangata-finance/mangata-e2e/assets/15139384/3b0ed480-7c5b-4a2b-bae8-cdb2e95f66ca)

Reply:
Fixed